### ISSUE TITLE
Removing circular references

### ISSUE THREAD

**Issue by ichorid** [Tribler/tribler#3896]
Circular references in objects structure is bad practice for [many reasons](https://softwareengineering.stackexchange.com/questions/11856/whats-wrong-with-circular-references). The worst result of this practice is a [God-object anti-pattern](https://en.wikipedia.org/wiki/God_object). We actually have a whole Trinity of God-objects in LaunchManyCore, Session, and DownloadManager. 

These anti-patterns significantly complicate mocking, reduce testing efficiency, and prevent us from achieving a truly modular design, that is the first step to [the microservice architecture](https://github.com/Tribler/py-ipv8/issues/123).

In general, functions/methods should never *extract* information from its arguments. For example, we want the class `__init__` arguments list to represent the minimal set of things that this class requires to be instantiated and do its thing.

In short, we must refactor code like this:

```python

class SomeModule():
    def __init__(self, session):
        self.session = session
   
    def do_something(self): 
        print(self.session.some_property)
        print(self.session.another_property)
```
to

``` python
class SomeModule():
    def __init__(self, property, another_property):
        self.property = property
        self.another_property = another_property
   
    def do_something(self): 
        print(self.some_property)
        print(self.another_property)
```

We are already going to [refactor the libtorrent wrapper](#3418), so this is a suitable moment to add this rule of no parent referencing to Tribler coding standards.
